### PR TITLE
accounts/abi: accounts/abi/bind: Move topics to abi package

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
The topic functionality is imo better suited for the abi package than the bind package.
Additional: more tests, slight changes in bind and templates (reduce duplicated logic).

### CODE REVIEW

File: accounts/abi/bind/bind.go [hunk 1]
```diff
@@ -589,73 +587,12 @@ func hasStruct(t abi.Type) bool {
...
-	}
-	identity := fmt.Sprintf("function %v", method.RawName)
-	if method.Type == abi.Fallback {
-		identity = "fallback"
-	} else if method.Type == abi.Receive {
-		identity = "receive"
-	}
-	return fmt.Sprintf("%s(%v) %sreturns(%v)", identity, strings.Join(inputs, ", "), state, strings.Join(outputs, ", "))
+func formatMethod(method abi.Method) string {
+	return method.String()
```
Comment (gballet) [inline accounts/abi/bind/bind.go:592]: how about removing `formatMethod` and `formatEvent` altogether if they are simply an alias for `.String()` ?

**Comment by gballet** [comment]
@MariusVanDerWijden topics functions are only used in the `bind` package, what is your reasoning for moving them to abi?

File: accounts/abi/bind/bind.go (hunk 1 above)
Comment (MariusVanDerWijden) [reply accounts/abi/bind/bind.go:592]: yes!

**Comment by MariusVanDerWijden** [comment]
@gballet All abi functions are currently only used in the bind package. My understanding is, that we use `ABI.Unpack()` for unindexed arguments and `parseTopics` for indexed arguments. I would like to unify these types of conversion in the future.

**Pull request merged**